test/provisioner/provision.go

```diff
@@ -114,6 +158,143 @@ func GetInstallOverlay(provider string, installDir string) (InstallOverlay, erro
+	filePath := "base/kustomization.yaml"
+	content, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		fmt.Printf("Error reading kustomization file: %v\n", err)
+		return err
+	}
+
+	// Convert the content to a string.
+	kustomizationContent := string(content)
+
+	// Define the values to update.
+	kustomizationContent = strings.Replace(kustomizationContent, "newName: ghcr.io/confidential-containers/key-broker-service", "newName: "+imagePath, -1)
+	kustomizationContent = strings.Replace(kustomizationContent, "newTag: built-in-as-v0.7.0", "newTag: "+imageTag, -1)
+
+	// Write the updated content back to the same file.
+	err = ioutil.WriteFile(filePath, []byte(kustomizationContent), 0644)
+	if err != nil {
+		fmt.Printf("Error writing to kustomization file: %v\n", err)
+		return err
+	}
+
+	fmt.Println("Kustomization file updated successfully.")
+	return nil
+
+}
+
+func (p *KeyBrokerService) Deploy(ctx context.Context, imagePath string, imageTag string) error {
+	originalDir, err := os.Getwd()
+	if err != nil {
+		fmt.Printf("Error getting the current working directory: %v\n", err)
+		return err
+	}
+
+	// jump to kbs kubernetes config directory
+	newDirectory := "kbs/config/kubernetes/"
+	err = os.Chdir(newDirectory)
+	if err != nil {
+		fmt.Printf("Error changing the working directory: %v\n", err)
+		return err
+	}
+
+	// Note: Use kustomize overlay to update this
+	err = UpdateKbsKustomizationFile(imagePath, imageTag)
+	if err != nil {
+		fmt.Printf("Error updating kustomization file: %v\n", err)
+		return err
+	}
+
+	// Deploy kbs
+	k8sCnfDir, err := os.Getwd()
+	if err != nil {
+		fmt.Printf("Error getting the current working directory: %v\n", err)
+		return err
+	}
+	fmt.Println(k8sCnfDir)
+
+	keyFile := filepath.Join(k8sCnfDir, "overlays/key.bin")
+	if _, err := os.Stat(keyFile); os.IsNotExist(err) {
+		fmt.Println("key.bin file does not exist")
+		//return err
+	}
+
+	kbsCert := filepath.Join(k8sCnfDir, "base/kbs.pem")
+	if _, err := os.Stat(kbsCert); os.IsNotExist(err) {
+		kbsKey := filepath.Join(k8sCnfDir, "base/kbs.key")
+		keyOutputFile, err := os.Create(kbsKey)
+		if err != nil {
+			fmt.Printf("Error creating key file: %v\n", err)
+			os.Exit(1)
+		}
+		defer keyOutputFile.Close()
+
+		opensslGenPKeyCmd := exec.Command("openssl", "genpkey", "-algorithm", "ed25519")
+		opensslGenPKeyCmd.Stdout = keyOutputFile
+		opensslGenPKeyCmd.Stderr = os.Stderr
+		fmt.Printf("Running command: %v\n", opensslGenPKeyCmd.Args)
+		if err := opensslGenPKeyCmd.Run(); err != nil {
+			fmt.Printf("Error generating key: %v\n", err)
+			return err
+		}
+
+		opensslPKeyCmd := exec.Command("openssl", "pkey", "-in", kbsKey, "-pubout", "-out", kbsCert)
+		opensslPKeyCmd.Stdout = os.Stdout
+		opensslPKeyCmd.Stderr = os.Stderr
+		if err := opensslPKeyCmd.Run(); err != nil {
+			fmt.Printf("Error creating kbs.pem: %v\n", err)
+			return err
+		}
+	}
+
+	kubectlApplyCmd := exec.Command("kubectl", "apply", "-k", k8sCnfDir+"/overlays")
```

Comment:
All of this can be done using kustomization libraries, isn't it? That's how we deploy CAA for provisioner, no?